source/common/http/http3/conn_pool.cc

```diff
@@ -155,8 +154,7 @@ allocateConnPool(Event::Dispatcher& dispatcher, Random::RandomGenerator& random_
                             "Creating Http/3 client");
         // If there's no ssl context, the secrets are not loaded. Fast-fail by returning null.
         auto factory = &pool->host()->transportSocketFactory();
-        ASSERT(dynamic_cast<Quic::QuicClientTransportSocketFactory*>(factory) != nullptr);
-        if (static_cast<Quic::QuicClientTransportSocketFactory*>(factory)->sslCtx() == nullptr) {
+        if (factory->sslCtx() == nullptr) {
```

Comment:
There is a warning log following this line. Does it still indicate an unexpected behavior now?

Reply:
I assume later code will hard fail if the sslCtx is null - I just changed how we accessed it.